src/views/award.ts

```diff
@@ -14,54 +14,68 @@
  * limitations under the License.
  */
 
-import { AwardNode } from '@manuscripts/transform'
+import { ContextMenu, ContextMenuProps } from '@manuscripts/style-guide'
+import { AwardNode, schema } from '@manuscripts/transform'
 
-import { Trackable } from '../types'
+import { AwardModal, AwardModalProps } from '../components/awards/AwardModal'
+import {
+  DeleteAwardDialog,
+  DeleteAwardDialogProps,
+} from '../components/awards/DeleteAwardDiaolog'
+import { isDeleted } from '../lib/track-changes-utils'
+import { updateNodeAttrs } from '../lib/view'
+import { Trackable, TrackableAttributes } from '../types'
 import BlockView from './block_view'
 import { createNodeView } from './creators'
+import ReactSubView from './ReactSubView'
 
+export type AwardAttrs = TrackableAttributes<AwardNode>
 export class AwardView extends BlockView<Trackable<AwardNode>> {
-  public initialise() {
-    this.createDOM()
-    this.contentDOM = this.dom
-    this.updateContents()
-  }
+  protected popperContainer: HTMLDivElement
+  private dialog: HTMLElement
 
   public updateContents() {
     if (!this.contentDOM) {
       return
     }
-
-    this.contentDOM.innerHTML = ''
-    this.contentDOM.classList.remove('block-container')
-    const notAvailable = 'N/A'
-    const { recipient, code, source } = this.node.attrs
+    const { id, recipient, code, source } = this.node.attrs
     if (!source) {
       return
     }
 
+    this.contentDOM.className = 'block block-award'
+    this.contentDOM.setAttribute('id', id)
+    this.contentDOM.innerHTML = ''
+    const notAvailable = 'N/A'
     const fragment = document.createDocumentFragment()
 
-    fragment.appendChild(this.createAwardElement('award-source', source))
+    fragment.appendChild(
+      this.createAwardFragment('award-source', `${source ? source : ''}`)
+    )
 
     fragment.appendChild(
-      this.createAwardElement(
+      this.createAwardFragment(
         'award-code',
         `Grant Number(s): ${code ? code.split(';').join(', ') : notAvailable}`
       )
     )
 
     fragment.appendChild(
-      this.createAwardElement(
+      this.createAwardFragment(
         'award-recipient',
         `Recipient: ${recipient ? recipient : notAvailable}`
       )
     )
+    if (this.props.getCapabilities().editArticle) {
+      this.dom.addEventListener('mouseup', this.handleClick)
+    }
 
     this.contentDOM.appendChild(fragment)
+
+    this.handleTrackChanges()
   }
 
-  private createAwardElement = (
+  private createAwardFragment = (
```

Comment:
Can we make this more structured? something like:
```
<div class="award-code">
  <span class="title">Grant Number(s):</span>
  <span class="content">...</span>
</div>
```

Reply:
Done